src/python/pants/goal/run_tracker.py

```diff
@@ -286,13 +286,16 @@ def end_workunit(self, workunit):
             self.self_timings.add_timing(path, self_time, is_tool)
             self.outcomes[path] = workunit.outcome_string(workunit.outcome())
 
+    def get_cumulative_timings(self) -> TimingData:
+        return self.cumulative_timings.get_all()  # type: ignore[no-any-return]
+
     def get_critical_path_timings(self):
         """Get the cumulative timings of each goal and all of the goals it (transitively) depended
         on."""
         setup_workunit = WorkUnitLabel.SETUP.lower()
         transitive_dependencies = dict()
         raw_timings = dict()
-        for entry in self.cumulative_timings.get_all():
+        for entry in self.get_cumulative_timings():
             raw_timings[entry["label"]] = entry["timing"]
```

Comment:
I think that this method is dead: goals don't have dependencies anymore.

Reply:
good point added https://github.com/pantsbuild/pants/pull/11387.
will rebase/merge this PR if https://github.com/pantsbuild/pants/pull/11387 is accepted and merged.